translator/trace/jaeger/protospan_to_jaegerproto.go

```diff
@@ -187,7 +187,10 @@ func ocLinksToJaegerReferencesProto(ocSpanLinks *tracepb.Span_Links) ([]jaeger.S
 		var traceID jaeger.TraceID
 		// Checks for nil/wrongLen traceID are already done in ocSpansToJaegerSpansProto()
 		traceIDHigh, traceIDLow, err := tracetranslator.BytesToUInt64TraceID(ocLink.TraceId)
-		if err != nil {
+		if err == tracetranslator.ErrNilTraceID {
+			traceIDLow = 0
+			traceIDHigh = 0
```

Comment:
Does this only apply to jaeger? If we are fine with this behavior, may be this function should not return an error at all and accept 0 as a valid ID?

Reply:
I had considered this, but `BytesToInt64TraceID` and `BytesToUInt64TraceID` are also used to translate the TraceID on the span itself and I did not want to adjust this behavior.

I'm honestly not sure what the Jaeger pipeline would do with a span with a 0 trace or span ID.

Reply:
I think we should adopt this behavior universally for all translation code (not just Jaeger).

If we receive a span with 0 ID we should pass it as is (if we can). It is up to Jaeger to decide how they deal with a 0 ID span that they receive from the Collector. Collector is not intended to be a sanitizer of telemetry data _by default_. The default behavior should be to pass everything as is, without modifying anything unless the modification cannot be avoided. This should be the default behavior and a design goal for the Collector. It is not Collector's business to interpret and modify data _unless_ the end user specifically configures it to do so using processors.

If sanitizing of 0 ID is desirable then it can be implemented as a processor.

Reply:
Agree with Tigran. That's what I was leading to as well.

Reply:
Thanks for the direction.  PR has been updated.  See the latest comment.